### PR TITLE
Use TLD for PLD code

### PR DESCRIPTION
**Pull request by kkrugler**
Trivial change to switch the PaidLevelDomain code to use EffectiveTldFinder

### CODE REVIEW

File: src/main/java/crawlercommons/domains/PaidLevelDomain.java [hunk 1]
```diff
@@ -64,60 +48,20 @@ public static String getPLD(String hostname) {
...
-                // Must be xxx.de.com format
-                firstHostPiece = numPieces - 3;
-            } else {
-                // Must be xxx.com format
-                firstHostPiece = numPieces - 2;
-            }
+        // Now use support in EffectiveTldFinder
+        String result = EffectiveTldFinder.getAssignedDomain(hostname, true);
+        if (result == null) {
+        	LOGGER.debug(String.format("Hostname %s isn't a valid FQDN", hostname));
```
Comment (jnioche) [inline src/main/java/crawlercommons/domains/PaidLevelDomain.java]: You don't need to use StringFormat with slf4j: just do 
`("Hostname {} isn't a valid FQDN", hostname)`
also the call to String.format is causing the build to fail, see https://travis-ci.org/crawler-commons/crawler-commons/builds/293871975#L1048

Comment (kkrugler) [reply src/main/java/crawlercommons/domains/PaidLevelDomain.java]: Hi @jnioche - Yeah, I've got projects with log4j and slf4j, I keep forgetting to use the slf4j formatting support. Also thanks for the heads-up re forbidden apis, I didn't realize I needed to run `mvn forbiddenapis:check` to catch these problems - I though `mvn package` did that.

Comment (kkrugler) [reply src/main/java/crawlercommons/domains/PaidLevelDomain.java]: Fixed

Comment (jnioche) [reply src/main/java/crawlercommons/domains/PaidLevelDomain.java]: Hi @kkrugler, I had forgotten about forbiddenapis but that shows that the automated checks with PRs are very useful. I'll have a closer look at the PR on Tuesday. Are you happy with me releasing 0.9 before merging it? we could release 0.9.1 shortly after that

Comment (kkrugler) [reply src/main/java/crawlercommons/domains/PaidLevelDomain.java]: Yes, please push out 0.9 - this is a very low priority change, just something I noticed in some code using the PLD support.

File: src/main/java/crawlercommons/domains/EffectiveTldFinder.java [hunk 2]
```diff
@@ -308,13 +310,14 @@ private static String join(String[] ary) {
         public EffectiveTLD(String line) {
             if (line.startsWith(EXCEPTION)) {
                 exception = true;
-                domain = line.substring(1, line.length());
+                domain = line.substring(EXCEPTION.length(), line.length());
```
Comment (sebastian-nagel) [inline src/main/java/crawlercommons/domains/EffectiveTldFinder.java]: +1

**Comment by kkrugler** [comment]
Hi @sebastian-nagel - you're right, the PLD code should explicitly exclude private domains. Would it work for me to commit this, and then you make that change as part of your pull request #185?

**Comment by sebastian-nagel** [comment]
Addressed @kkrugler's comment in #185: however, this requires a change in unit tests: `de.com` is part of the private section in the public suffixes list, hence `xxx.de.com` isn't a pay-level domain.

**Pull request merged**